### PR TITLE
lshw: unstable-2023-03-20 -> 02.20

### PR DESCRIPTION
**Pull request by thiagokokada**
## Description of changes

- Migrate to `by-name`
- Bump: unstable-2023-03-20 -> 02.20 (actually called B02.20 upstream, but we have a workaround because limitations in our Repology export)
- ~~Enable GUI by default (and fix it)~~

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by thiagokokada** [comment]
Result of `nixpkgs-review pr 279592` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>deepin.deepin-clone</li>
  </ul>
</details>
<details>
  <summary>58 packages built:</summary>
  <ul>
    <li>deepin.dde-app-services</li>
    <li>deepin.dde-appearance</li>
    <li>deepin.dde-application-manager</li>
    <li>deepin.dde-calendar</li>
    <li>deepin.dde-clipboard</li>
    <li>deepin.dde-control-center</li>
    <li>deepin.dde-control-center.dev</li>
    <li>deepin.dde-daemon</li>
    <li>deepin.dde-device-formatter</li>
    <li>deepin.dde-dock</li>
    <li>deepin.dde-dock.dev</li>
    <li>deepin.dde-file-manager</li>
    <li>deepin.dde-gsettings-schemas</li>
    <li>deepin.dde-launchpad</li>
    <li>deepin.dde-network-core</li>
    <li>deepin.dde-polkit-agent</li>
    <li>deepin.dde-qt-dbus-factory</li>
    <li>deepin.dde-session</li>
    <li>deepin.dde-session-shell</li>
    <li>deepin.dde-session-shell.dev</li>
    <li>deepin.dde-session-ui</li>
    <li>deepin.dde-widgets</li>
    <li>deepin.deepin-album</li>
    <li>deepin.deepin-calculator</li>
    <li>deepin.deepin-camera</li>
    <li>deepin.deepin-compressor</li>
    <li>deepin.deepin-draw</li>
    <li>deepin.deepin-editor</li>
    <li>deepin.deepin-image-viewer</li>
    <li>deepin.deepin-movie-reborn</li>
    <li>deepin.deepin-movie-reborn.dev</li>
    <li>deepin.deepin-music</li>
    <li>deepin.deepin-picker</li>
    <li>deepin.deepin-reader</li>
    <li>deepin.deepin-screen-recorder</li>
    <li>deepin.deepin-screensaver</li>
    <li>deepin.deepin-shortcut-viewer</li>
    <li>deepin.deepin-system-monitor</li>
    <li>deepin.deepin-terminal</li>
    <li>deepin.deepin-turbo</li>
    <li>deepin.deepin-voice-note</li>
    <li>deepin.dpa-ext-gnomekeyring</li>
    <li>deepin.dtkcore</li>
    <li>deepin.dtkcore.dev</li>
    <li>deepin.dtkcore.doc</li>
    <li>deepin.dtkdeclarative</li>
    <li>deepin.dtkdeclarative.dev</li>
    <li>deepin.dtkdeclarative.doc</li>
    <li>deepin.dtkgui</li>
    <li>deepin.dtkgui.dev</li>
    <li>deepin.dtkgui.doc</li>
    <li>deepin.dtkwidget</li>
    <li>deepin.dtkwidget.dev</li>
    <li>deepin.dtkwidget.doc</li>
    <li>deepin.image-editor</li>
    <li>deepin.qt5integration</li>
    <li>lshw</li>
    <li>webex</li>
  </ul>
</details>

File: pkgs/by-name/ls/lshw/package.nix [hunk 1]
```diff
@@ -11,24 +11,22 @@
...
-  # FIXME: when switching to a stable release:
   # Fix repology.org by not including the prefixed B, otherwise the `pname` attr
   # gets filled as `lshw-B.XX.XX` in `nix-env --query --available --attr nixpkgs.lshw --meta`
   # See https://github.com/NixOS/nix/pull/4463 for a definitive fix
-  version = "unstable-2023-03-20";
+  version = "02.20";
 
   src = fetchFromGitHub {
     owner = "lyonel";
     repo = pname;
```
Comment (AndersonTorres) [inline pkgs/by-name/ls/lshw/package.nix:21]: https://github.com/NixOS/nixpkgs/pull/278611

**Review by AndersonTorres — APPROVED** [review]
Except by pname, in L21, LGTM

File: pkgs/by-name/ls/lshw/package.nix (hunk 1 above)
Comment (thiagokokada) [reply pkgs/by-name/ls/lshw/package.nix:21]: From the [tracking issue](https://github.com/NixOS/nixpkgs/issues/277994) it seems usage of `pname` in `rec` in fine, so I will ignore this issue for now until we have a conclusion for that issue.

Comment (thiagokokada) [reply pkgs/by-name/ls/lshw/package.nix:21]: BTW, I will eventually rewrite my derivations to use `finalAttrs`. But today is not this day.

**Pull request merged**